firestore-send-email/functions/src/index.ts

```diff
@@ -343,17 +408,19 @@ async function deliver(
       response: result.response || null,
     };
 
+    // Update Firestore document to indicate success
     update["delivery.state"] = "SUCCESS";
     update["delivery.info"] = info;
 
     logs.delivered(ref, info);
   } catch (e) {
+    // Update Firestore document to indicate failure
     update["delivery.state"] = "ERROR";
     update["delivery.error"] = e.toString();
     logs.deliveryError(ref, e);
   }
 
-  // Wrapping in transaction to allow for automatic retries (#48)
+  // Update the Firestore document transactionally to allow retries
```

Comment:
keep old comment as it's referencing an issue

Reply:
Added issue to new comment.